feat: add long polling REST API limitation (#6799)
camunda/camunda-docs: docs/apis-tools/java-client/job-worker.md

The Java client provides a job worker that handles polling for available jobs. This allows you to focus on writing code to handle the activated jobs.

:::caution REST API limitation

> The Java client cannot reliably keep the long-lived polling connections required for job polling via the Camunda 8 REST API.

Author: It is when the cases mentioned below occur ([original issue description](https://github.com/camunda/camunda-docs/issues/4375)). I've updated the sentence to remove the ambiguity a bit more.

- Receiving additional job activation requests from a Java client running on the same JVM while the long-polling connection is still open.
- Receiving additional job activation requests from a Java client running on a different JVM while the long-polling connection is still open.

> Additionally, the long-polling connection may still receive jobs after the Java client has been closed.
> As the Java client will not process these jobs, they will time out, leading to some jobs not being processed and becoming available again after their timeout has elapsed.

Reviewer: ```suggestion
Additionally, the long-polling connection might still receive jobs after the Java client is closed.
As the Java client does not process these jobs they will time out. This means some jobs are not processed and will become available again after their timeout has elapsed.
```
